hs_file_types/models/base.py

```diff
@@ -954,11 +954,13 @@ def supports_unzip(self):
     @property
     def aggregation_name(self):
         """Returns aggregation name as per the aggregation naming rule defined in issue#2568"""
+        if self.is_fileset:
+            return self.folder
         short_path = self.files.first().short_path
         if "/" in short_path:
             short_path = short_path.rsplit("/")[0]
             return os.path.join(short_path, self.dataset_name)
```

Comment:
What happens if two aggregations in same path location have same dataset_name?

Reply:
That's a valid point and it also applies to us displaying the dataset_name as the virtual folder in the ui.  Perhaps we could enforce unique dataset_names for aggregations in the same folder.  What do you think?

Reply:
Seems like we would have to do this. It's a lot like ensuring unique file names in a folder - which people are already familiar with and so from a user perspective if we have to alert them that they can't have two aggregations with the same dataset name in the same folder I think they would get it.

Reply:
Making dataset_name unique within same file path location is good from UI perspective - less user confusion. However, using dataset_name as part of the aggregation_name probably would require quite a bit of code changes including fixing tons of unit tests again.  If the dataset_name is needed for aggregation download, then it may be better to implement a new method/property in ```AbstractLogicalFile``` class:

```
    @property
    def dataset_name_path(self):
        """ used for identifying aggregation for aggregation download only"""
```

Reply:
yeah, changing aggregation_name to use the dataset_name has opened a can of worms.  If we use the dataset_name_path property you're suggesting, then we would still have to enforce unique dataset_names.  I need to think this over some more.

We could also add a query parameter that indicates the aggregation should be downloaded instead of just the primary file... maybe.

Reply:
Using an additional query parameter to indicate aggregation download probably be least amount of work. In that case we don't have to enforce uniqueness of dataset_name. If the user renames virtual folders to create duplicate dataset_name, user can always edit it to remove his confusion.